src/client/pages/EmailSent.tsx

```diff
@@ -40,49 +54,144 @@ export const EmailSent = ({
 	children,
 	formError,
 	instructionContext,
+	hasStateHandle,
+	passcodeAction,
+	fieldErrors,
+	passcode,
+	timeUntilTokenExpiry,
 }: PropsWithChildren<Props>) => {
 	const [recaptchaErrorMessage, setRecaptchaErrorMessage] = useState('');
 	const [recaptchaErrorContext, setRecaptchaErrorContext] =
 		useState<ReactNode>(null);
+
+	useEffect(() => {
+		// we only want this to run in the browser as window is not
+		// defined on the server
+		// and we also check that the expiry time exists so that
+		// we redirect to the session expired page
+		// if the token expires while the user is on the current page
+		if (typeof window !== 'undefined' && timeUntilTokenExpiry) {
+			logger.info(`Welcome page: loaded successfully`, undefined, {
+				timeUntilTokenExpiry,
+			});
+			setTimeout(() => {
+				logger.info(
+					`Welcome page: redirecting to token expired page`,
+					undefined,
+					{ timeUntilTokenExpiry },
+				);
+				window.location.replace(buildUrl('/register/code/expired'));
+			}, timeUntilTokenExpiry);
+		}
+	}, [timeUntilTokenExpiry]);
+
+	// autofocus the code input field when the page loads
+	useEffect(() => {
+		if (typeof window !== 'undefined') {
+			const codeInput: HTMLInputElement | null =
+				window.document.querySelector('input[name="code"]');
+
+			if (codeInput) {
+				codeInput.focus();
+			}
+		}
+	}, []);
+
+	const pageHeader = hasStateHandle
+		? 'Enter your code'
+		: 'Check your email inbox';
+
 	return (
 		<MainLayout
-			pageHeader="Check your email inbox"
+			pageHeader={pageHeader}
 			successOverride={showSuccess ? 'Email sent' : undefined}
 			errorOverride={
 				recaptchaErrorMessage ? recaptchaErrorMessage : errorMessage
 			}
 			errorContext={recaptchaErrorContext}
 		>
 			{children}
-			{email ? (
-				<MainBodyText>
-					We’ve sent an email to <b>{email}</b>
-				</MainBodyText>
-			) : (
-				<MainBodyText>We’ve sent you an email.</MainBodyText>
+			{!hasStateHandle && (
+				<>
+					{email ? (
+						<MainBodyText>
+							We’ve sent an email to <b>{email}</b>
+						</MainBodyText>
+					) : (
+						<MainBodyText>We’ve sent you an email.</MainBodyText>
+					)}
+					{instructionContext ? (
+						<MainBodyText>
+							Please follow the link in the email to {instructionContext}.
+						</MainBodyText>
+					) : (
+						<MainBodyText>
+							Please follow the instructions in this email.
+						</MainBodyText>
+					)}
+				</>
 			)}
-			{instructionContext ? (
-				<MainBodyText>
-					Please follow the link in the email to {instructionContext}.
-				</MainBodyText>
-			) : (
-				<MainBodyText>
-					Please follow the instructions in this email.
-				</MainBodyText>
+			{hasStateHandle && (
+				<>
+					{email ? (
+						<MainBodyText>
+							We’ve sent an email to <b>{email}</b> with verification
```

Comment:
This is odd wording ('verification instructions and a verification code') - surely just 'with a verification code' is fine?

Reply:
I'll speak to @nickhartyyy about this

Reply:
also thinking about this, if we get this merged in, it's easier to get opinions and feedback on wording/structure which is easier to change afterwards anyway